src/metabase/api/collection.clj

```diff
@@ -152,17 +159,19 @@
 
   This endpoint is intended to power a 'Root Folder View' for the Current User, so regardless you'll see all the
   top-level objects you're allowed to access."
-  [model archived]
-  {model    (s/maybe (s/enum "card" "dashboard" "pulse" "collection"))
-   archived (s/maybe su/BooleanString)}
+  [model archived writeable]
+  {model     (s/maybe (s/enum "card" "dashboard" "pulse" "collection"))
+   archived  (s/maybe su/BooleanString)
+   writeable (s/maybe su/BooleanString)}
   ;; Return collection contents, including Collections that have an effective location of being in the Root
   ;; Collection for the Current User.
   (collection-items
     collection/root-collection
-    {:model     (if (mi/can-read? collection/root-collection)
-                  (keyword model)
-                  :collection)
-     :archived? (Boolean/parseBoolean archived)}))
+    {:model      (if (mi/can-read? collection/root-collection)
+                   (keyword model)
+                   :collection)
+     :archived?  (Boolean/parseBoolean archived)
+     :writeable? (Boolean/parseBoolean writeable)}))
```

Comment:
If `writeable` is false, will this filter to just collections that aren't writeable or all collections?

Reply:
yeah good question, I struggled with a good name. Another possibility was `permission-level`, maybe that'd be clearer?

Reply:
maybe another name would be `must-be-writeable`. I think it'd be clearer that if `must-be-writeable` is false, that wouldn't mean to exclude writeable items